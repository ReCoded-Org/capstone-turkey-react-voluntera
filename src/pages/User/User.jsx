import { useState } from 'react';
import { useSelector } from 'react-redux';
import moment from 'moment';

import UsersProjectsCard from '../../components/UsersProjectsCard';
import AddNewProjectModal from '../../components/AddNewProjectModal';

function User() {
  const user = useSelector((state) => state.user.user);
  const [isClick, setisClick] = useState(false);

  return (
    <div className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8">
      <div className="grid grid-cols-1 gap-5 py-5 ">
        {/* user personal details col */}
        <div className=" py-2 pl-2 grid lg:grid-cols-3 md:grid-cols-3 sm:grid-cols-3 grid-cols-1 gap-4 items-center place-items-center ">
          <div className="col-span-2 order-1 sm:order-none px-5">
            <h1 className="text-blue-500 text-2xl py-2 lg:text-3xl">
              {user?.firstName} {user?.lastName}
            </h1>
            <div className="text-md">
              <p>
                <strong>Username: </strong> {user?.username}
              </p>
              <p>
                <strong>User Type: </strong> {user?.userType}
              </p>
              <p>
                <strong>Date of Birth: </strong>{' '}
                {moment(user?.birthDate).utc().format('MM/DD/YYYY')}
              </p>
              <p>
                <strong>email: </strong> {user?.email}
              </p>
              <p>
                <strong>phone: </strong> {user?.phone || 'N/A'}
              </p>
              {user?.address && (
                <p>
                  <strong>address: </strong> {user?.address || 'N/A'}
                </p>
              )}
            </div>
            {/* skills */}
            <p className="mt-3 flex flex-wrap gap-1">
              <strong className="text-lg text-blue-500 mr-2">
                {user?.skills ? (
                  'SKILLS :'
                ) : (
                  <div>
                    Description:{' '}
                    <span className="text-slate-500 text-md font-normal">
                      {user?.description}
                    </span>
                  </div>
                )}
              </strong>
              {user?.skills?.map((skill) => {
                return (
                  <span
                    key={skill}
                    className="border-r-2 border-r-blue-400 pr-2 px-1"
                  >
                    {skill}
                  </span>
                );
              })}
            </p>
          </div>
          <div className="w-48 h-48 lg:w-52 lg:h-52 md:w-52 md:h-52 rounded-full  border-2 p-1 col-span-1">
            <img
              src="https://i.pinimg.com/280x280_RS/2e/45/66/2e4566fd829bcf9eb11ccdb5f252b02f.jpg"
              alt="somethine"
              className="rounded-full"
            />
          </div>
        </div>
        {/* add project */}
        {user?.userType === 'Organization' && (
          <button
            type="button"
            className=" px-4 py-1 border bg-purple-400 w-36 ml-7"
            onClick={() => setisClick(!isClick)}
          >
            Add project
          </button>
        )}
        <div className={isClick ? 'block' : 'hidden'}>
          <AddNewProjectModal />
        </div>
        {/* projects col */}
        <div className=" px-2 py-2 text-center">
          <h1 className="text-blue-500 text-2xl text-center mb-5  pb-2">
            Project Assigned
          </h1>
          <div className=" p-2 grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 md:grid-cols-3 gap-5 place-items-center items-center justify-center">
            {user?.projects?.map((card) => {
              return <UsersProjectsCard key={card} userdata={card} />;
            })}
          </div>
          <div className="max-w-7xl ">
            {!user?.projects?.length > 0 ? (
              <span className="text-3xl text-slate-700 flex justify-center">
                You have no projects assigned !
              </span>
            ) : null}
          </div>
        </div>
      </div>
    </div>
  );
}

export default User;
