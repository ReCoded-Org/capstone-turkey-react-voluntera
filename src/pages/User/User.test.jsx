import renderer from 'react-test-renderer';

import User from './User';

test('render the Userprofile page', () => {
  const tree = renderer.create(<User />).toJSON();
  expect(tree).toMatchSnapshot();
});
