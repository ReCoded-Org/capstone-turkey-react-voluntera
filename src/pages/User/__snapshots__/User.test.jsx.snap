// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Userprofile page 1`] = `
<div
  className="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
>
  <div
    className="grid grid-cols-1 gap-5 py-5 "
  >
    <div
      className=" py-2 pl-2 grid lg:grid-cols-3 md:grid-cols-3 sm:grid-cols-3 grid-cols-1 gap-4 items-center place-items-center "
    >
      <div
        className="col-span-2 order-1 sm:order-none px-5"
      >
        <h1
          className="text-blue-500 text-2xl py-2 lg:text-3xl"
        >
          Tom Holland
        </h1>
        <div
          className="text-md"
        >
          <p>
            <strong>
              Username: 
            </strong>
             username34
          </p>
          <p>
            <strong>
              User Type: 
            </strong>
             Volunteer / Organisation
          </p>
          <p>
            <strong>
              Date of Birth: 
            </strong>
             2022-03-10
          </p>
          <p>
            <strong>
              age: 
            </strong>
             25
          </p>
          <p>
            <strong>
              email: 
            </strong>
             user@gmail.com
          </p>
          <p>
            <strong>
              phone: 
            </strong>
             +90 (321) 123456
          </p>
        </div>
        <p
          className="mt-3 flex flex-wrap gap-1"
        >
          <strong
            className="text-lg text-blue-500 mr-2"
          >
            SKILLS: 
          </strong>
          <span
            className="border-r-2 border-r-blue-400 pr-2 px-1"
          >
            Problem Soving
          </span>
          <span
            className="border-r-2 border-r-blue-400 pr-2 px-1"
          >
            Internet Knowledge
          </span>
          <span
            className="border-r-2 border-r-blue-400 pr-2 px-1"
          >
            Teamwork
          </span>
          <span
            className="border-r-2 border-r-blue-400 pr-2 px-1"
          >
            Organising Management
          </span>
        </p>
      </div>
      <div
        className="w-48 h-48 lg:w-52 lg:h-52 md:w-52 md:h-52 rounded-full  border-2 p-1 col-span-1"
      >
        <img
          alt="somethine"
          className="rounded-full"
          src="https://www.pngitem.com/pimgs/m/75-758282_walter-circle-person-photo-in-circle-hd-png.png"
        />
      </div>
    </div>
    <div
      className=" px-2 py-2"
    >
      <h1
        className="text-blue-500 text-2xl text-center mb-5  pb-2"
      >
        Project Assigned
      </h1>
      <div
        className=" p-2 grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-4 md:grid-cols-3 gap-5 place-items-center items-center"
      >
        <div
          className="p-6 lg:max-w-sm md:max-w-sm max-w-xs bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700"
        >
          <a
            href="http://www.w3.org/2000/svg"
          >
            <h5
              className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
            >
              New technology 2022
            </h5>
          </a>
          <p
            className="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
             Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <a
            className="inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-purple-700 rounded-lg hover:bg-purple-800 focus:ring-4 focus:outline-none focus:ring-purple-300 dark:bg-purple-600 dark:hover:bg-purple-700 dark:focus:ring-purple-800"
            href="http://www.w3.org/2000/svg"
          >
            Read more
            <svg
              className="ml-2 -mr-1 w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fillRule="evenodd"
              />
            </svg>
          </a>
        </div>
        <div
          className="p-6 lg:max-w-sm md:max-w-sm max-w-xs bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700"
        >
          <a
            href="http://www.w3.org/2000/svg"
          >
            <h5
              className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
            >
              New technology 2022
            </h5>
          </a>
          <p
            className="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
             Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <a
            className="inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-purple-700 rounded-lg hover:bg-purple-800 focus:ring-4 focus:outline-none focus:ring-purple-300 dark:bg-purple-600 dark:hover:bg-purple-700 dark:focus:ring-purple-800"
            href="http://www.w3.org/2000/svg"
          >
            Read more
            <svg
              className="ml-2 -mr-1 w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fillRule="evenodd"
              />
            </svg>
          </a>
        </div>
        <div
          className="p-6 lg:max-w-sm md:max-w-sm max-w-xs bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700"
        >
          <a
            href="http://www.w3.org/2000/svg"
          >
            <h5
              className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
            >
              New technology 2022
            </h5>
          </a>
          <p
            className="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
             Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <a
            className="inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-purple-700 rounded-lg hover:bg-purple-800 focus:ring-4 focus:outline-none focus:ring-purple-300 dark:bg-purple-600 dark:hover:bg-purple-700 dark:focus:ring-purple-800"
            href="http://www.w3.org/2000/svg"
          >
            Read more
            <svg
              className="ml-2 -mr-1 w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fillRule="evenodd"
              />
            </svg>
          </a>
        </div>
        <div
          className="p-6 lg:max-w-sm md:max-w-sm max-w-xs bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700"
        >
          <a
            href="http://www.w3.org/2000/svg"
          >
            <h5
              className="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
            >
              New technology 2022
            </h5>
          </a>
          <p
            className="mb-3 font-normal text-gray-700 dark:text-gray-400"
          >
             Here are the biggest enterprise technology acquisitions of 2021 so far, in reverse chronological order.
          </p>
          <a
            className="inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-purple-700 rounded-lg hover:bg-purple-800 focus:ring-4 focus:outline-none focus:ring-purple-300 dark:bg-purple-600 dark:hover:bg-purple-700 dark:focus:ring-purple-800"
            href="http://www.w3.org/2000/svg"
          >
            Read more
            <svg
              className="ml-2 -mr-1 w-4 h-4"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clipRule="evenodd"
                d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                fillRule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
